### PR TITLE
Affix webserver access_denied warning to be configurable

### PR DESCRIPTION
**Pull request by amoghrajesh**
Add a possibility to make the airflow webserver's access denied message - which occurs when an user tries to perform some operation outside their user rights. Allow this to be configurable through config so that every webserver installation can have a custom message without changing the airflow code.

Note: This message is not per call to the `has_access` function, but per webserver installation which is a more realistic case.

Took suggestions from community and approached the problem accordingly.
- Made a function `get_access_denied_message` so that the "unit" is testable.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/www/auth.py
Comment (uranusjr) [inline airflow/www/auth.py:32]: This should either use `get_mandatory` or provide a fallback.

Comment (potiuk) [reply airflow/www/auth.py:32]: Not really. The `get_mandatory` is needed only when we want to have a value but the value has no default (for example when we want to get a URL that has no default but we **really** need it to be configured). 

The way it works when it is put in config.yml is that "default" is automatically used as fallback if there is no fallback (It's been working like that already for quite some time via defaul_config.cfg  - so my recent change has not changed it - jut made it works without the default_config.cfg

Comment (amoghrajesh) [reply airflow/www/auth.py:32]: I don't think it is needed as @potiuk mentions here. The older default text is passed as the default in the new property in config.yml, which is basically the fallback.

File: airflow/config_templates/config.yml
Comment (pankajkoti) [inline airflow/config_templates/config.yml]:
```suggestion
        The notification that appears when a user attempts to execute actions beyond their authorised privileges.
```

File: airflow/config_templates/config.yml
Comment (pankajkoti) [inline airflow/config_templates/config.yml:1358]:
```suggestion
      default: "Unauthorized Access: Denied"
```

**Review by pankajkoti — APPROVED** [review]
Looks good to me, minor cosmetic suggestions if you'd like to consider :)

File: airflow/config_templates/config.yml
Comment (vincbeck) [reply airflow/config_templates/config.yml:1358]: I dont think we can do that. The current message is "Access is Denied" so in order to keep the current experience, we got to have this default value

**Review by vincbeck — APPROVED** [review]

File: airflow/config_templates/config.yml
Comment (amoghrajesh) [reply airflow/config_templates/config.yml]: I would like to keep it as message because this is not a notification. Maybe I can term it to warning instead

File: airflow/config_templates/config.yml
Comment (amoghrajesh) [reply airflow/config_templates/config.yml:1358]: Agreed. We do not want to break the backward compatibility experience. If we want to make this change, we should have a different task for it instead

File: airflow/config_templates/config.yml
Comment (pankajkoti) [reply airflow/config_templates/config.yml]: okay so 
"`The message displayed when a user attempts to execute actions beyond their authorised privileges.`"
?

Comment (pankajkoti) [reply airflow/config_templates/config.yml]: If not, the minimum suggestion is to correct `an user` to `a user` 🙂

Comment (amoghrajesh) [reply airflow/config_templates/config.yml]: Yeah why not, thanks, making the changes

**Review by pankajkoti — APPROVED** [review]

File: airflow/config_templates/config.yml
Comment (hussein-awala) [reply airflow/config_templates/config.yml:1358]: I agree with @vincbeck

File: tests/www/test_security.py
Comment (hussein-awala) [inline tests/www/test_security.py]: could you replace the `conf.get("webserver", "access_denied_message")` by `get_access_denied_message` method?
And maybe separate them in two tests: 
- test_default_access_denied_message
- test_custom_access_denied_message

Comment (amoghrajesh) [reply tests/www/test_security.py]: Yeah why not. Very valid suggestions. Incorporated them.

**Review by hussein-awala — APPROVED** [review]

**Comment by hussein-awala** [comment]
Waiting for a second look from @uranusjr before merge

**Review by potiuk — APPROVED** [review]

**Review by aritra24 — APPROVED** [review]

**Review by eladkal — APPROVED** [review]
Lets merge so it won't miss 2.7.0 
we can modify in followup PR if needed

**Pull request merged**
